Assets/Scripts/SS3D/Systems/Chat/ChatWindow.cs

```diff
@@ -0,0 +1,179 @@
+        {
+            base.OnDisabled();
+
+            _controls.SendChatMessage.performed += HandleSendMessage;
+            ChatMessageSender.RegisterChatWindow(this);
+        }
+
+        protected override void OnDisabled()
+        {
+            base.OnDisabled();
+
+            _controls.SendChatMessage.performed -= HandleSendMessage;
+            ChatMessageSender.UnregisterChatWindow(this);
+        }
+
+        public void OnPointerEnter(PointerEventData eventData)
+        {
+            _inputSystem.ToggleBinding("<Mouse>/scroll/y", false);
+        }
+
+        public void OnPointerExit(PointerEventData eventData)
+        {
+            _inputSystem.ToggleBinding("<Mouse>/scroll/y", true);
+        }
+
+        protected virtual void HandleSendMessage(InputAction.CallbackContext context)
+        {
+            SendMessage();
+        }
+
+        public void SendMessage()
+        {
+            string text = inputField.text;
+            if (text.Length <= 0)
+            {
+                return;
+            }
+            
+            inputField.text = "";
+            
+            ChatChannel chatChannel = GetCurrentChatChannel();
+            if (chatChannel == null)
+            {
+                return;
+            }
+            
+            PlayerSystem playerSystem = Subsystems.Get<PlayerSystem>();
+            string playerCkey = playerSystem.GetCkey(InstanceFinder.ClientManager.Connection);
+            Player player = playerSystem.GetPlayer(playerCkey);
+
+            if (availableChannels.Contains(chatChannel.name))
+            {
+                ChatMessageSender.SendPlayerMessage(chatChannel, text, player);
+            }
+            else
+            {
+                ChatMessageSender.SendServerMessageToCurrentPlayer(
+                    chatChannels.GetChannelForInGameChatSystemMessages.name, 
+                    $"[UNAUTHORIZED ACCESS TO {chatChannel.name} CHANNEL]");
+            }
+        }
+
+        protected void ShowMessages(List<ChatMessage> messages)
+        {
+            StringBuilder sb = new StringBuilder();
+            foreach (ChatMessage message in messages)
+            {
+                ChatChannel chatChannel = chatChannels.GetChannels().First(x => x.name == message.channel);
+
+                sb.Append(chatChannel.prefix);
+                
+                sb.AppendFormat("<color=#{0}>", ColorUtility.ToHtmlStringRGBA(chatChannel.color));
+                
+                if (!string.IsNullOrEmpty(chatChannel.abbreviation))
+                {
+                    sb.AppendFormat("[{0}] ", chatChannel.abbreviation);
+                }
+                
+                if (!chatChannel.hideSenderName)
+                {
+                    sb.AppendFormat("{0}: " , message.sender);
+                }
+                
+                sb.AppendFormat("{0}\n", message.text);
+                
+                sb.Append(chatChannel.suffix);
```

Comment:
You can move all this logic to a helper function in ChatMessage, just create a ChatMessage.ToString() override function and make your custom logic in there. Also, with a chat system you can create a getter for the channel and not make a Linq call each time, just create a Dictionary<string, ChatChannel> RegisteredChatChannels to get that info.